### ISSUE TITLE
mirror-command sets wrong client_count if frame is accessed by index

### ISSUE THREAD

**Issue by astier** [herbstluftwm/herbstluftwm#1151]
## Info

herbstluftwm 0.9.1 (285f0f54 built on 2021-01-21)

## Problem

The mirror-command swaps two frames including the windows contained in those frames (which is the correct behavior). However, after executing the mirror-command the attribute `client_count` of both frames is wrong if its accessed by index with the following command (in this example index 0):

```sh
herbstclient get_attr tags.focus.tiling.root.0.client_count
```

However, if frame with index 0 is focused and the client_count is accessed by `focused_frame` then the client_count is correct:

```sh
herbstclient get_attr tags.focus.tiling.focused_frame.client_count
```

## Steps to reproduce

1. Create two frames. One left with index 0 and one right with index 1.
2. Open two terminals in frame 0 and one terminal in frame 1.
3. Verify the index and client_count of both frames as instructed below.
4. Mirror both frames with `herbstclient mirror`
5. Verify the index and client_count of both frames again as instructed below

It can be seen that the client_count for frame 0 is still 2 and for frame 1 its still 1 if accessed by index, although it should be reversed.

## client_count verification

Follow this steps once for each frame to verify the frame-index and `client_count`. `idx` refers in the following to the index of the `focused_frame`.

```sh
herbstclient get_attr tags.focus.tiling.focused_frame.index
herbstclient get_attr tags.focus.tiling.focused_frame.client_count
herbstclient get_attr tags.focus.tiling.root.idx.client_count
```

**Comment by t-wissmann** [comment]
Thanks @astier, for the excellent bug report! It's even the issue that the entire frame objects are not swapped correctly, sorry for the trouble!

I've fixed it in PR #1152.

**Comment by astier** [comment]
It works now. Thanks.
